### PR TITLE
PARQUET-1070 Add CPack support to the build

### PR DESCRIPTION
**Pull request by trink**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,10 +18,25 @@
...
+if(NOT VERSION)
+  message(FATAL_ERROR "invalid .parquetcppversion")
+endif()
+
+project(parquet-cpp VERSION ${VERSION})
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "C++ library to read and write the Apache Parquet columnar data format")
+set(CPACK_PACKAGE_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
+set(CPACK_PACKAGE_VERSION_MINOR ${PROJECT_VERSION_MINOR})
+set(CPACK_PACKAGE_VERSION_PATCH ${PROJECT_VERSION_PATCH})
+set(CPACK_PACKAGE_VENDOR        "Apache")
```
Comment (wesm) [inline CMakeLists.txt]: Apache Software Foundation

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,10 +18,25 @@
...
+set(CPACK_PACKAGE_CONTACT       "Dev <dev@parquet.apache.org>")
```
Comment (wesm) [inline CMakeLists.txt]: maybe "Apache Parquet Development" ?

File: cmake_modules/parquet-cppConfig.cmake.in [hunk 3]
```diff
@@ -0,0 +1,15 @@
+set(PARQUET-CPP_VERSION x.y.z)
```
Comment (wesm) [inline cmake_modules/parquet-cppConfig.cmake.in]: is this file path the typical style (i.e. would it be obvious to other CMake users what is in here)?

Comment (trink) [reply cmake_modules/parquet-cppConfig.cmake.in]: Yes, see https://cmake.org/cmake/help/v3.5/module/CMakePackageConfigHelpers.html?highlight=cmakepackageconfighelpers
This allows find_package to be used in config mode i.e.,
`find_package(parquet-cpp 1.3.0 REQUIRED CONFIG)`

**Comment by wesm** [comment]
+1, thank you!

**Pull request closed without merging**